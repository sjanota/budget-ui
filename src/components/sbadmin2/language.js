import React, { createContext, useContext } from 'react';
import { useSBAdmin2 } from './context';
import PropTypes from 'prop-types';
import { capitalize } from '../../util/capitalize';

const DictionaryContext = createContext();

export function DictionaryProvider({ dictionaries, children }) {
  const { user } = useSBAdmin2();
  return (
    <DictionaryContext.Provider value={dictionaries[user.locale]}>
      {children}
    </DictionaryContext.Provider>
  );
}

DictionaryProvider.propTypes = {
  dictionaries: PropTypes.object,
  children: PropTypes.node,
};

export const useDictionary = () => useContext(DictionaryContext);

export function withDictionary(prop, Component, readPropName) {
  return props => {
    readPropName = readPropName || `read${capitalize(prop)}`;
    const readDict = props[readPropName];
    const newProps = { ...props };
    delete newProps[readPropName];
    const dictionary = useDictionary();
    if (readDict) {
      newProps[prop] = readDict(dictionary);
    }
    return <Component {...newProps} />;
  };
}
