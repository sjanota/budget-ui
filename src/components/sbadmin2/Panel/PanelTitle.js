import React from 'react';
import PropTypes from 'prop-types';
import { withDictionary } from '../language';

function PanelTitle({ title }) {
  return <h6 className="m-0 font-weight-bold text-primary">{title}</h6>;
}

PanelTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PanelTitle);
