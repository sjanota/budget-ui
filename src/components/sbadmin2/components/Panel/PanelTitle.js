import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withDictionary } from '../../language';

function PanelTitle({ children, title, className }) {
  const classNames = classnames(
    'm-0',
    'font-weight-bold',
    'text-primary',
    className
  );
  return (
    <h6 className={classNames}>
      {children}
      {title}
    </h6>
  );
}

PanelTitle.propTypes = {
  title: PropTypes.node.isRequired,
  className: PropTypes.string,
  children: PropTypes.node,
};

export default withDictionary('title', PanelTitle);
