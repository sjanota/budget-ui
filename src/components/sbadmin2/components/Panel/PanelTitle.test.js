import React from 'react';
import renderer from 'react-test-renderer';
import PanelTitle from './PanelTitle';
import { mount } from 'enzyme';
import { DictionaryContext } from '../../language';

describe('PanelTitle', () => {
  it('uses provided with title', () => {
    const component = mount(<PanelTitle title="My page" />);
    expect(component).toHaveText('My page');
  });

  it('reads title from dictionary', () => {
    const component = mount(
      <DictionaryContext.Provider value={{ title: 'My page' }}>
        <PanelTitle readTitle={d => d.title} />
      </DictionaryContext.Provider>
    );
    expect(component).toHaveText('My page');
  });

  it('renders with minimal props', () => {
    const component = renderer.create(<PanelTitle title="" />);
    expect(component).toMatchSnapshot();
  });
});
