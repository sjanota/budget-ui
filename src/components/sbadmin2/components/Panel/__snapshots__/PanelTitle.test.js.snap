// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelTitle renders with minimal props 1`] = `
<h6
  className="m-0 font-weight-bold text-primary"
>
  
</h6>
`;
